### ISSUE TITLE
TreeList - New Filter Modes

### ISSUE THREAD

**Issue by Alyar666** [DevExpress/DevExtreme#7270]
# The Problem

In the TreeList widget, filter results include matching rows with their ancestors but without descendants.

# The Proposed Solution

We added the `filterMode` option that controls which rows filter and search results should include.

This option accepts the following values:

- `fullBranch`
The results include rows that meet the filter condition and their ancestors and descendants. The descendant rows appear collapsed.

- `withAncestors` (default mode)
The results include rows that meet the filter condition and their ancestors.

The following images illustrate these modes:

```js
let treeListOptions = {
    filterMode: "fullBranch"
};
```
![fullBranch](https://user-images.githubusercontent.com/23191430/54284437-f8aa9400-45b0-11e9-8cc2-336adfe38fe8.png)

```js
let treeListOptions = {
    filterMode: "withAncestors"
};
```
![withAncestors](https://user-images.githubusercontent.com/23191430/54284451-fea07500-45b0-11e9-9c01-bbc9bef5701b.png)

# Try It

## Live Sandbox

[jQuery](https://codepen.io/anon/pen/JzrQKb)

[Angular](https://codesandbox.io/s/xo8y304ljp)

## Installation

Follow the installation guide from the [release page](https://github.com/DevExpress/DevExtreme/releases) and leave us feedback on this topic.

# We Need Your Feedback

## Take a Quick Poll

[Do you find these filter modes useful?](https://docs.google.com/forms/d/e/1FAIpQLSf0NojqtOVz8hlPgOa8EgUG-s9u6hVilG1yIYrYTnCzKJMQhw/viewform)

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by hakimio** [comment]
Can we also get this feature for TreeView?

**Comment by dxbykov** [comment]
Hi @hakimio,

> Can we also get this feature for TreeView?

It's not in our plans for v19.1, but we'll consider possibilities of adding this functionality to our TreeView. BTW, can it be an option for you to replace your TreeView with a single-column and headerless TreeList?

**Comment by hakimio** [comment]
> can it be an option for you to replace your TreeView with a single-column and headerless TreeList?

Hi Alexander,
Yes, that's an option. The only real drawback is that it would increase the bundle size significantly.

**Comment by dxbykov** [comment]
Am I right that you are not using TreeList or DataGrid in your app?

**Comment by hakimio** [comment]
> Am I right that you are not using TreeList or DataGrid in your app?

Only DataGrid.

**Comment by hakimio** [comment]
Just made a quick test: importing TreeList module to Angular app, adds ~65KB to production bundle (before gzip), while TreeView module only adds ~2KB.

**Comment by dxbykov** [comment]
With all dependencies our TreeList shares about 90-95% of its code with DataGrid. So, according to my estimation, adding a TreeList to your app will increase the bundle size roughly by 30K (gzipped).

**Comment by spaatz880** [comment]
What about adding child filtering to GridView?

**Comment by Alyar666** [comment]
Hello @spaatz880,

What do you mean by children in the GridView?

**Comment by genachka** [comment]
This is a great feature, thank you for adding it!

**Comment by dxbykov** [comment]
Thank you to everybody who gave us feedback on this feature. It's available in the [v19.1](https://github.com/DevExpress/DevExtreme/releases/tag/19.1.3) release. I'm closing this thread. In the case of bugs or questions, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
